pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/ChannelManager.java

```diff
@@ -41,9 +43,11 @@ public class ChannelManager {
 
   private final ConcurrentHashMap<String, ManagedChannel> _channelMap = new ConcurrentHashMap<>();
 
-  public ChannelManager(GrpcMailboxService mailboxService, PinotConfiguration extraConfig) {
+  public ChannelManager(GrpcMailboxService mailboxService, PinotConfiguration extraConfig,
+      Consumer<MailboxIdentifier> gotMailCallback) {
     _mailboxService = mailboxService;
-    _grpcMailboxServer = new GrpcMailboxServer(_mailboxService, _mailboxService.getMailboxPort(), extraConfig);
+    _grpcMailboxServer = new GrpcMailboxServer(
+        _mailboxService, _mailboxService.getMailboxPort(), extraConfig, gotMailCallback);
```

Comment:
this looks like a design issue of the code structure. ChannelManager doesn't need to know about the mailbox callback. 
it seems like the reason is that the member variable: `GRPCMailboxServer` needs to know about the gotMailCallback in order to create the StreamObserver correctly. 

can't we add an API in the `GRPCMailboxService.getMailCallback()` it should be registered at the MailboxService level IMO b/c you should never have different gotMailCallback on the same MailboxService (or am I not correct on this assumption?)

Reply:
I see what you are saying and I think this makes sense, since this isn't a minor code refactoring (most of the PR is just piping this through) - do you mind if I do that as a follow-up?

Reply:
sounds good